### PR TITLE
detect URLs in Instructions field and add as attachment in FACE assignmnts

### PR DESCRIPTION
**Pull request by spandanabaruah**
- doesn't currently name the attachment based on the URL (that will come in the next commits)

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js [hunk 1]
```diff
@@ -78,6 +82,23 @@ class AssignmentEditorDetail extends ErrorHandlingMixin(SaveStatusMixin(EntityMi
 		}
 	}
 
+	_addToCollection(attachment) {
```
Comment (mdgbayly) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js]: minor nit: Does having this as a separate method add much value. I guess most of the time we are just adding a single link, but if we do add multiple links, e.g. through pasting, then it means that we are finding the attachment collection for every link. So we could just load the collection directly in the addLinks method?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js [hunk 2]
```diff
@@ -78,6 +82,23 @@ class AssignmentEditorDetail extends ErrorHandlingMixin(SaveStatusMixin(EntityMi
...
+		const collection = attachmentCollectionStore.get(this._attachmentsHref);
+		collection.addAttachment(attachment);
+	}
+
+	addLinks(links) {
```
Comment (mdgbayly) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js]: nit: should be a private method `_addLinks`?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js [hunk 3]
```diff
@@ -4,28 +4,33 @@ import '../d2l-activity-due-date-editor.js';
...
 import { timeOut } from '@polymer/polymer/lib/utils/async.js';
 
 class AssignmentEditorDetail extends ErrorHandlingMixin(SaveStatusMixin(EntityMixinLit(LocalizeMixin(RtlMixin(ActivityEditorMixin(MobxLitElement)))))) {
 
 	static get properties() {
 		return {
 			_nameError: { type: String },
-			_attachmentsHref: { type: String }
+			_attachmentsHref: { type: String },
+			_linksProcessor: { type: Object }
```
Comment (mdgbayly) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js]: nit: doesn't need to be a Lit property. It's just an internal property. We don't expect it's value to affect rendering.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js (hunk 1 above)
Comment (spandanabaruah) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js]: good point! I'll just load the collection in the addLinks method :)

Comment (spandanabaruah) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js]: Fixed it in the next PR for naming the attachments

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js [hunk 4]
```diff
@@ -45,8 +50,6 @@ class AssignmentEditorDetail extends ErrorHandlingMixin(SaveStatusMixin(EntityMi
 				#score-and-duedate-container {
 					display: flex;
 					flex-wrap: wrap;
-					min-height: 90px;  /* Hack to force a consistent the height for the old */
-					padding-bottom: 0; /* datetime picker. Can hopefully be removed when the new picker is used.*/
 				}
```
Comment (jchapmanD2L) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js:53]: @spandanabaruah was this change intentional?  I think it might have reverted some stuff @emil-furniss did to make the date picker stay in place.

Comment (spandanabaruah) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js:53]: oops! - not sure how I ended up deleting those lines. Will add them back.

Comment (spandanabaruah) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.js:53]: merged the PR - updating activities in BSI now
